pyanaconda/modules/subscription/satellite.py

```diff
@@ -0,0 +1,176 @@
+            return None
+
+
+def run_satellite_provisioning_script(provisioning_script=None, run_on_target_system=False):
+    """Run the Satellite provisioning script.
+
+    Each Satellite instance provides a provisioning script that will
+    enable the currently running environment to talk to the given
+    Satellite instance.
+
+    This means that the self-signed certificates of the given
+    Satellite instance will be installed to the system but also some
+    necessary changes will be done to rhsm.conf.
+
+    As we need to provision both the installation *and* target system
+    to talk to Satellite we need to run the provisioning script twice,
+    once in the installation environment and once on the target system.
+
+    This is achieved by running this function first in the installation environment
+    with run_on_target_system == False before a registration attempt.
+    And then in the installation phase with run_on_target_system == True.
+
+    Implementation wise we just always run the script from a tempfile.
+
+    That way we can easily run it in the installation environment as well
+    as in the target system chroot with minimum code needed to make sure
+    it exists where we need it
+
+    :param str provisioning_script: content of the Satellite provisioning script
+                                    or None if no script is available
+    :param str run_on_target_system: run in the target system chroot instead,
+                                     otherwise run in the installation environment
+    :return: True on success, False otherwise
+    :rtype: bool
+    """
+    # first check we actually have the script
+    if provisioning_script is None:
+        log.warning("subscription: satellite provisioning script not available")
+        return False
+
+    # now that we have something to run, check where to run it
+    if run_on_target_system:
+        # run in the target system chroot
+        sysroot = conf.target.system_root
+    else:
+        # run in installation environment
+        sysroot = "/"
+
+    # create the tempfile containing the script in the sysroot in /tmp, just in case
+    sysroot_tmp = util.join_paths(sysroot, "/tmp")
+    with tempfile.NamedTemporaryFile(mode="w+t", dir=sysroot_tmp, prefix="satellite-") as tf:
+        # write the provisioning script to the tempfile
+        tf.write(provisioning_script)
+        # We always set root to the correct sysroot, so the script will always
+        # look like it is in /tmp. So just split the randomly generated file name
+        # and combine it with /tmp to get sysroot specific script path.
+        filename = os.path.split(tf.name)[1]
+        chroot_script_path = os.path.join("/tmp", filename)
+        # and execute it in the sysroot
+        rc = util.execWithRedirect("bash", argv=[chroot_script_path], root=sysroot)
+        if rc == 0:
+            log.debug("subscription: satellite provisioning script executed successfully")
```

Comment:
Could you please check journal and make sure that the new logging messages make sense? I think there will be many similar lines, that don't say anything new.

Reply:
I've just checked and IMHO it seems reasonably verbose - we can always reduce the verbosity later if all turns out to work fine but for now I would like to keep it as it is.